include/aws/common/json.h

```diff
@@ -0,0 +1,395 @@
+#ifndef AWS_COMMON_JSON_H
+#define AWS_COMMON_JSON_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include "thread.h"
+
+struct aws_json_node;
+
+// ====================
+// Create and pass type
+
+/**
+ * Creates a new aws_json_node and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * @return A new aws_json_node
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_node_new(void);
```

Comment:
[trivial] the offical json terminology is ["value"](https://www.json.org/json-en.html). we might want to use that instead of "node"